test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -853,497 +1072,306 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+            tlsCertificateKeyFile: process.env.KMIP_TLS_CERT_FILE
           }
         }
+      };
+      clientWithTlsExpiredOptions = {
+        keyVaultNamespace,
+        kmsProviders: getKmsProviders(null, '127.0.0.1:8000', '127.0.0.1:8000', '127.0.0.1:8000'),
+        tlsOptions: tlsCaOptions
+      };
+      clientWithInvalidHostnameOptions = {
+        keyVaultNamespace,
+        kmsProviders: getKmsProviders(null, '127.0.0.1:8001', '127.0.0.1:8001', '127.0.0.1:8001'),
+        tlsOptions: tlsCaOptions
+      };
+      clientNoTls = this.configuration.newClient({}, { autoEncryption: clientNoTlsOptions });
+      clientWithTls = this.configuration.newClient({}, { autoEncryption: clientWithTlsOptions });
+      clientWithTlsExpired = this.configuration.newClient(
+        {},
+        { autoEncryption: clientWithTlsExpiredOptions }
+      );
+      clientWithInvalidHostname = this.configuration.newClient(
+        {},
+        { autoEncryption: clientWithInvalidHostnameOptions }
+      );
+      const mongodbClientEncryption = this.configuration.mongodbClientEncryption;
+      clientEncryptionNoTls = new mongodbClientEncryption.ClientEncryption(clientNoTls, {
+        ...clientNoTlsOptions,
+        bson: BSON
+      });
+      clientEncryptionWithTls = new mongodbClientEncryption.ClientEncryption(clientWithTls, {
+        ...clientWithTlsOptions,
+        bson: BSON
+      });
+      clientEncryptionWithTlsExpired = new mongodbClientEncryption.ClientEncryption(
+        clientWithTlsExpired,
+        { ...clientWithTlsExpiredOptions, bson: BSON }
+      );
+      clientEncryptionWithInvalidHostname = new mongodbClientEncryption.ClientEncryption(
+        clientWithInvalidHostname,
+        { ...clientWithInvalidHostnameOptions, bson: BSON }
       );
-
-      return this.clientEncrypted.connect();
     });
 
-    afterEach(function () {
-      return this.clientEncrypted && this.clientEncrypted.close();
-    });
+    // Case 1.
+    context('Case 1: AWS', metadata, function () {
+      const masterKey = {
+        region: 'us-east-1',
+        key: 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0',
+        endpoint: '127.0.0.1:8002'
+      };
+      const masterKeyExpired = { ...masterKey, endpoint: '127.0.0.1:8000' };
+      const masterKeyInvalidHostname = { ...masterKey, endpoint: '127.0.0.1:8001' };
 
-    it('should error when inserting into a view with autoEncryption', metadata, function () {
-      return this.clientEncrypted
-        .db(dataDbName)
-        .collection('view')
-        .insertOne({ a: 1 })
-        .then(
-          () => {
-            throw new Error('Expected insert to fail, but it succeeded');
-          },
-          err => {
-            expect(err)
-              .to.have.property('message')
-              .that.matches(/cannot auto encrypt a view/);
-          }
-        );
+      it('fails with various invalid tls options', metadata, async function () {
+        try {
```

Comment:
Can we break apart this test into multiple `it` blocks?

Reply:
(same comment for the tests below)

Reply:
arguably out of scope since this is how they are already implemented and it could  mess with how beforeEach/afterEach gets applied, but I'll see what I can do

Reply:
updated